optaplanner-core/src/main/java/org/optaplanner/core/api/domain/solution/drools/ProblemFactCollectionProperty.java

```diff
@@ -40,10 +40,11 @@
  * Do not annotate {@link PlanningEntity planning entities} as problem facts:
  * they are automatically available as facts for {@link ConstraintFactory#from(Class)} or DRL.
  *
- * @see ProblemFactProperty
+ * @deprecated In favor of {@link org.optaplanner.core.api.domain.solution.ProblemFactCollectionProperty}.
  */
 @Target({ METHOD, FIELD })
 @Retention(RUNTIME)
+@Deprecated(/* forRemoval = true */)
 public @interface ProblemFactCollectionProperty {
```

Comment:
Do we have a testdata unit test that still uses these deprecated ProblemFactCollectionProperty? (Including a // TODO remove in 8.0)
Almost all of our users use this, many won't refactor quickly. We need to make sure this keeps working during the course of the 7.x time.